### ISSUE TITLE
Protected types inherited documentation does not extend template instantiation past first level of inheritance

### ISSUE THREAD

**Issue by scott-zhong** [doxygen/doxygen#10656]
Protected types inherited documentation does not extend template instantiation past first level of inheritance. The problem started with Doxygen 1.9.5 and continues with trunk 1.11.0 (https://github.com/doxygen/doxygen/commit/4c99c07d2eeaf5284a040ef7995e0998106f5063*).

```
/**
 * Guard docs
 */
template <class Resource>
class Guard
{
};

/**
 * Lock docs
 */
class Lock
{
};

/**
 * Base docs
 */
template <class T>
class Base
{
protected:
    /**
     * guard typedef
     */
    typedef Guard< Base< T > > LockGuard;
};

/**
 * Derived docs
 */
template <class T>
class Derived : public Base<T>
{
protected:
    /**
     * guard typedef
     */
    typedef Guard< Base<T> > LockGuard;
};

/**
 * NextGen docs
 */
class NextGen
    : protected Derived<Lock>
{
};
```

Doxygen 1.9.4:

![image](https://github.com/doxygen/doxygen/assets/32584126/841950c2-1eab-42d4-8372-840609cdc913)

Doxygen 1.9.5 and trunk (https://github.com/doxygen/doxygen/commit/4c99c07d2eeaf5284a040ef7995e0998106f5063*):

![image](https://github.com/doxygen/doxygen/assets/32584126/7873337d-52d7-48d4-8b24-cf232758c32a)

Testcase:

[testcase.zip](https://github.com/doxygen/doxygen/files/14288736/testcase.zip)

**Comment by albert-github** [comment]
A bisect gives:

db8c2d6289d8c56a401d07714a6109ddc0bdc819 is the first bad commit
```
commit db8c2d6289d8c56a401d07714a6109ddc0bdc819
Date:   Sun Jun 5 15:22:02 2022 +0200

    Performance tweaks to speed up processing

```

**Comment by albert-github** [comment]
Some observations.
When generating the textual CLASS_GRAPH (i.e `CLASS_GRAPH = TEXT`) with the 1.9.4 version we see:
![image](https://github.com/doxygen/doxygen/assets/5223533/23000503-03fb-4f2d-ac9d-044a7135d21f)

with the current master (1.11.0 (4c99c07d2eeaf5284a040ef7995e0998106f5063)) we see:
![image](https://github.com/doxygen/doxygen/assets/5223533/d0db9e30-00f2-45a1-b81c-3049e1cea645)


Regarding the graphical representation with the current master when `HAVE_DOT = YES` (and `CLASS_GRAPH = YES`):
![image](https://github.com/doxygen/doxygen/assets/5223533/7c5c8dba-06b2-48e3-8d0d-1a432fd77f8e)


and with `HAVE_DOT = NO`:
![image](https://github.com/doxygen/doxygen/assets/5223533/520dcaaf-daff-46d3-9602-63335e6181a6)

@doxygen any ideas?

**Comment by scott-zhong** [comment]
@doxygen the referenced commit fixes the problem. Thank you!!!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
